### DISCUSSION TITLE
Cleaning up repository in-memory

### DISCUSSION THREAD

**Discussion by kimdre** [go-git/go-git#1153] [Q&A] [answered]
I'm rather new to Go so please forgive me this weird question. 

I'm using memfs to clone a repo in-memory.

```go
func CloneRepository(url, ref string) (*git.Repository, error) {
	// Filesystem abstraction based on memory
	fs := memfs.New()
	// Git objects storer based on memory
	storer := memory.NewStorage()

	return git.Clone(storer, fs, &git.CloneOptions{
		URL:           url,
		SingleBranch:  true,
		ReferenceName: plumbing.ReferenceName(ref),
		Tags:          git.NoTags,
		Depth:         1,
	})
}
```

Do I have to consider anything when I want to remove the repo/filesystem in my memory, or is it as simple as just setting the variable to `nil`?

```go
repo, err := git.CloneRepository(cloneUrl, event.Ref)
if err != nil {
	log.Error("Failed to clone repository", logger.ErrAttr(err))
	return
}
repo = nil
```

**Comment by pjbgf** [comment, marked as answer]
@kimdre there is no need to set it to `nil`. Golang's garbage collector will take care of it and free up that memory once it goes "out of scope".

A few pointers on Go's GC:
- https://tip.golang.org/doc/gc-guide
- https://go.dev/blog/ismmkeynote

**Reply by kimdre** [reply]
Cool, thanks a lot :)
